src/main/java/net/sf/jabref/gui/auximport/FromAuxGroups.java

```diff
@@ -0,0 +1,291 @@
+
+        DefaultFormBuilder b = new DefaultFormBuilder(new FormLayout(
+                "left:pref, 4dlu, fill:pref:grow, 4dlu, left:pref", ""), buttons);
+        b.appendSeparator(Localization.lang("Options"));
+        b.append(Localization.lang("Reference database") + ":");
+        b.append(dbChooser, 3);
+        b.nextLine();
+        b.append(Localization.lang("LaTeX AUX file") + ":");
+        b.append(auxFileField);
+        b.append(browseAuxFileButton);
+        b.getPanel().setBorder(BorderFactory.createEmptyBorder(5, 5, 5, 5));
+
+        b = new DefaultFormBuilder(new FormLayout(
+                "fill:pref:grow, 4dlu, fill:pref:grow", "pref, pref, fill:pref:grow"), statusPanel);
+        b.appendSeparator(Localization.lang("Result"));
+        b.append(Localization.lang("Unknown BibTeX entries") + ":");
+        b.append(Localization.lang("Messages") + ":");
+        b.nextLine();
+        b.append(listScrollPane);
+        b.append(statusScrollPane);
+        b.getPanel().setBorder(BorderFactory.createEmptyBorder(5, 5, 5, 5));
+    }
+
+    private void parseActionPerformed() {
+        parseButton.setEnabled(false);
+        BasePanel bp = (BasePanel) parentTabbedPane.getComponentAt(dbChooser.getSelectedIndex());
+        notFoundList.removeAll();
+        statusInfos.setText(null);
+        BibDatabase refBase = bp.getDatabase();
+        String auxName = auxFileField.getText();
+
+        if ((auxName != null) && (refBase != null) && !auxName.isEmpty()) {
+            auxParser = new AuxParser(auxName, refBase);
+            AuxParserResult result = auxParser.parse();
+            notFoundList.setListData(result.getUnresolvedKeys().toArray(new String[result.getUnresolvedKeys().size()]));
+            statusInfos.append(result.getInformation(false));
+
+            generateButton.setEnabled(true);
+
+            // the generated database contains no entries -> no active generate-button
+            if (!result.getGeneratedBibDatabase().hasEntries()) {
+                statusInfos.append("\n" + Localization.lang("empty database"));
+                generateButton.setEnabled(false);
+            }
+        } else {
+            generateButton.setEnabled(false);
+        }
+
+        parseButton.setEnabled(true);
+
+        getGeneratedGroup();
+    }
+
+    public boolean generatePressed() {
+        return generatePressed;
+    }
+
+    public ExplicitGroup getGeneratedGroup() {
+        List<BibEntry> entries = auxParser.parse().getGeneratedBibDatabase().getEntries();
+        try {
+            ExplicitGroup group = new ExplicitGroup("SomePaperNameGroup", GroupHierarchyType.INDEPENDENT, Globals.prefs);
+            group.add(entries);
+            return group;
+        } catch  (ParseException e) {
+            return null;
+        }
+    }
+
+    /**
+     * Action used to produce a "Browse" button for one of the text fields.
+     */
+    static class BrowseAction extends AbstractAction {
```

Comment:
I think you can get rid of the Action Class and just use the actionPerformed method.